### PR TITLE
Create sol_vector_del_element function

### PR DESCRIPTION
**Pull request by otaviobp**
Add function to remove an element from vector. So far, this is usefull
for error handling in json.c file.

Function suggested by @lpereira in json review.

### CODE REVIEW

**Comment by barbieri** [comment]
+1, I could see lots of usage for that... grep for `sol_vector_del(&v, v.len - 1)` for many cases where we delete the last entry.

We also have that for ptr-vectors... in such case it's hard to find the index from the pointer, as you would need to give the pointer-to-pointer to avoid O(n) lookup. To these cases a simple 'del_last()` could help

File: src/lib/datatypes/include/sol-vector.h [hunk 1]
```diff
@@ -166,6 +167,12 @@ sol_ptr_vector_del(struct sol_ptr_vector *pv, uint16_t i)
     return sol_vector_del(&pv->base, i);
 }
 
+static inline int
+sol_ptr_vector_del_element(struct sol_ptr_vector *pv, const void *elem)
+{
+    return sol_vector_del_element(&pv->base, elem);
```
Comment (barbieri) [inline src/lib/datatypes/include/sol-vector.h]: this is not what I'd expect, as elem in a ptr-vector is the pointed-to... you'd need to walk the vector looking for elem.

File: src/lib/datatypes/sol-vector.c [hunk 2]
```diff
@@ -265,3 +265,19 @@ sol_ptr_vector_init_n(struct sol_ptr_vector *pv, uint16_t n)
...
+{
+    ssize_t offset;
+
+    offset = (const char *)elem - (const char *)v->data;
+    if (offset % v->elem_size)
+        return -EINVAL;
+
+    offset = offset / v->elem_size;
+    if (offset < 0 || offset >= v->len)
+        return -EINVAL;
```
Comment (barbieri) [inline src/lib/datatypes/sol-vector.c]: maybe -ENOENT?

File: src/test/test-vector.c [hunk 3]
```diff
@@ -360,4 +361,51 @@ vector_initializes_elements_to_zero(void)
 }
 
 
+DEFINE_TEST(test_vector_del_element);
+
+static void
+test_vector_del_element(void)
```
Comment (barbieri) [inline src/test/test-vector.c]: please also add a test to `sol_ptr_vector_del_element()`, I think your wrapper is not correct.

File: src/lib/datatypes/include/sol-vector.h (hunk 1 above)
Comment (otaviobp) [inline src/lib/datatypes/include/sol-vector.h]: That's true. I'll fix that and add the del_end you proposed too.

File: src/lib/datatypes/sol-vector.c (hunk 2 above)
Comment (otaviobp) [inline src/lib/datatypes/sol-vector.c]: I used the same return code from sol_vector_del when the index was invalid. But for this case maybe NOENT is better

File: src/test/test-vector.c (hunk 3 above)
Comment (otaviobp) [inline src/test/test-vector.c]: ok

**Pull request closed without merging**